scene/animation/animation_player.cpp

```diff
@@ -299,6 +299,10 @@ bool AnimationPlayer::_blend_pre_process(double p_delta, int p_track_count, cons
 		return false;
 	}
 
+	if (playback.current.from->animation == nullptr) {
+		playback.current.from = &animation_set[playback.assigned];
+	}
+
```

Comment:
I think it is rather necessary to avoid processing like:
```cpp
bool AnimationPlayer::_blend_pre_process(double p_delta, int p_track_count, const HashMap<NodePath, int> &p_track_map) {
	if (!playback.current.from || !playback.current.from->animation) {
		_set_process(false);
		return false;
	}
```

Reply:
Had to make a small adjustment
```
bool AnimationPlayer::_blend_pre_process(double p_delta, int p_track_count, const HashMap<NodePath, int> &p_track_map) {
if (!playback.current.from || playback.current.from->animation == nullptr) {
    _set_process(false);
    return false;
}
```

Instead of a crash know the animation just won't play. But at least it's not a crash. Maybe there should be a new issue for that? The current PR doesn't do that but I understand it's not preferable if we don't use it.

Reply:
Taking it to the extreme, I think it is not recommended to play the animation in `_ready()` in the editor. The playback you are trying to do in `_ready()` is implemented in core as an option called `autoplay`, which intentionally prevents autoplay in editor.

```cpp
void AnimationPlayer::_notification(int p_what) {
	switch (p_what) {
		case NOTIFICATION_READY: {
			if (!Engine::get_singleton()->is_editor_hint() && animation_set.has(autoplay)) { /// <- !Engine::get_singleton()->is_editor_hint() makes to avoid calling it on the Editor
				set_active(active);
				play(autoplay);
				_check_immediately_after_start();
			}
		} break;
	}
}
```

Probably this was already present before I rewrote AnimationPlayer.

However, upon closer investigation, there is already a line in `seek_internal()` that does almost exactly what this modification is doing now.

Considering that `_blend_pre_process()` is executed after `seek_internal(()`, I think this should only be done in one place. So I think the final fix could be something like this:

AnimationPlayer.h
```cpp
class AnimationPlayer : public AnimationMixer {
...
private:
...
	void _play(const StringName &p_name, double p_custom_blend = -1, float p_custom_scale = 1.0, bool p_from_end = false);
	void _capture(const StringName &p_name, bool p_from_end = false, double p_duration = -1.0, Tween::TransitionType p_trans_type = Tween::TRANS_LINEAR, Tween::EaseType p_ease_type = Tween::EASE_IN);
	void _process_playback_data(PlaybackData &cd, double p_delta, float p_blend, bool p_seeked, bool p_internal_seeked, bool p_started, bool p_is_current = false);
	void _blend_playback_data(double p_delta, bool p_started);
	bool _ensure_current_animation(); /// <- Add this.
	void _stop_internal(bool p_reset, bool p_keep_state);
	void _check_immediately_after_start();
```

AnimationPlayer.cpp
```cpp
bool AnimationPlayer::_ensure_current_animation() {
	if (!playback.current.from || playback.current.from->animation.is_null()) {
		if (playback.assigned) {
			ERR_FAIL_COND_V_MSG(!animation_set.has(playback.assigned), false, vformat("Animation not found: %s.", playback.assigned));
			playback.current.from = &animation_set[playback.assigned];
		}
		if (!playback.current.from) {
			return false; // There is no animation.
		}
	}
	return true;
}

bool AnimationPlayer::_blend_pre_process(double p_delta, int p_track_count, const HashMap<NodePath, int> &p_track_map) {
	if (!_ensure_current_animation()) {
		_set_process(false);
		return false;
	}

	if (playback.current.from->animation.is_null()) {
		playback.current.from = &animation_set[playback.assigned];
		if (!playback.current.from) {
			_set_process(false);
			return false;
		}
	}
...

void AnimationPlayer::seek_internal(double p_time, bool p_update, bool p_update_only, bool p_is_internal_seek) {
	if (!active) {
		return;
	}

	bool is_backward = Animation::is_less_approx(p_time, playback.current.pos);

	_check_immediately_after_start();

	playback.current.pos = p_time;
//	if (!playback.current.from || playback.current.from->animation.is_null()) {
//		if (playback.assigned) {
//			ERR_FAIL_COND_MSG(!animation_set.has(playback.assigned), vformat("Animation not found: %s.", playback.assigned));
//			playback.current.from = &animation_set[playback.assigned];
//		}
//		if (!playback.current.from) {
//			return; // There is no animation.
//		}
//	}
// Remove them.
	playback.seeked = true;
	playback.internal_seeked = p_is_internal_seek;

	if (p_update) {
		_process_animation(is_backward ? -0.0 : 0.0, p_update_only);
		playback.seeked = false; // If animation was proceeded here, no more seek in internal process.
	}
}
```

I am concerned about the possibility of new problems that might arise from the fact that the previous stop at `!playback.current.from` will now be followed by a ensure check there. However, since `current.from` is basically nulled when `stop()` is performed, I do not think this will be a concern.

However, this fix did not completely eliminate the crash. It surely no longer 100% crashes, but it crashes rarely when switching selection of the SceneTree between AnimationPlayer and other Node with switching scene, so it is possible that null access is occurring elsewhere which is what this is trying to fix.